### PR TITLE
📖 Add Azure tab for deploy Calico CNI step in quickstart

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**:  See https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/504

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -484,6 +484,9 @@ kubectl --namespace=default get secret/capi-quickstart-kubeconfig -o jsonpath={.
 
 Calico is used here as an example.
 
+{{#tabs name:"tab-deploy-cni" tabs:"AWS,Azure,Docker,GCP,vSphere,OpenStack,Metal3"}}
+{{#tab AWS|Docker|GCP|vSphere|OpenStack|Metal3}}
```
Comment (CecileRobertMichon) [inline docs/book/src/user/quick-start.md:488]: Not sure if these instructions apply to all the other providers, let me know if some of them shouldn't be listed here

**Comment by CecileRobertMichon** [comment]
/hold 

until https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/504 merges

File: docs/book/src/user/quick-start.md (hunk 1 above)
Comment (nader-ziada) [reply docs/book/src/user/quick-start.md:488]: Does it make sense to add a line about why Azure requires something different than the standard Calico? Not sure if its needed

Comment (CecileRobertMichon) [reply docs/book/src/user/quick-start.md:488]: done

**Comment by nader-ziada** [comment]
lgtm

**Pull request merged**